monitoring/prober/test_subscription_validation.py

```diff
@@ -0,0 +1,113 @@
+                  },
+                  'altitude_lo': 20,
+                  'altitude_hi': 400,
+              },
+              'time_start': time_start.strftime(common.DATE_FORMAT),
+              'time_end': time_end.strftime(common.DATE_FORMAT),
+          },
+          'callbacks': {
+              'identification_service_area_url': 'https://example.com/foo'
+          },
+      })
+  assert resp.status_code == 400
+
+
+def test_create_sub_missing_footprint(session, sub2_uuid):
+  time_start = datetime.datetime.utcnow()
+  time_end = time_start + datetime.timedelta(seconds=10)
+
+  resp = session.put(
+      '/subscriptions/{}'.format(sub2_uuid),
+      json={
+          'extents': {
+              'spatial_volume': {
+                  'altitude_lo': 20,
+                  'altitude_hi': 400,
+              },
+              'time_start': time_start.strftime(common.DATE_FORMAT),
+              'time_end': time_end.strftime(common.DATE_FORMAT),
+          },
+          'callbacks': {
+              'identification_service_area_url': 'https://example.com/foo'
+          },
+      })
+  assert resp.status_code == 400
+
+
+def test_create_sub_with_huge_area(session, sub2_uuid):
+  time_start = datetime.datetime.utcnow()
+  time_end = time_start + datetime.timedelta(seconds=10)
+
+  resp = session.put(
+      '/subscriptions/{}'.format(sub2_uuid),
+      json={
+          'extents': {
+              'spatial_volume': {
+                  'footprint': {
+                      'vertices': common.HUGE_VERTICES,
+                  },
+                  'altitude_lo': 20,
+                  'altitude_hi': 400,
+              },
+              'time_start': time_start.strftime(common.DATE_FORMAT),
+              'time_end': time_end.strftime(common.DATE_FORMAT),
+          },
+          'callbacks': {
+              'identification_service_area_url': 'https://example.com/foo'
+          },
+      })
+  assert resp.status_code == 400
+
+
+def test_create_too_man_subs(session):
+  time_start = datetime.datetime.utcnow()
+  time_end = time_start + datetime.timedelta(seconds=1)
+  all_resp = []
+
+  # create 1 more than the max allowed Subscriptions per area
+  for index in range(common.MAX_SUB_PER_AREA + 1):
+    resp = session.put(
+        '/subscriptions/{}'.format(str(uuid.uuid4())),
+        json={
+            'extents': {
+                'spatial_volume': {
+                    'footprint': {
+                        'vertices': common.VERTICES,
+                    },
+                    'altitude_lo': 20,
+                    'altitude_hi': 400,
+                },
+                'time_start': time_start.strftime(common.DATE_FORMAT),
+                'time_end': time_end.strftime(common.DATE_FORMAT),
+            },
+            'callbacks': {
+                'identification_service_area_url': 'https://example.com/foo'
+            },
+        })
+    all_resp.append(resp.status_code == 429)
+
+  assert any(all_resp)
```

Comment:
Probably better to assert `resp.status_code == (429 if index == common.MAX_SUB_PER_AREA else 200)` on each loop iteration because we want to make sure the first MAX_SUB_PER_AREA calls succeed in addition to the final one failing.  If this test passes with the current code then something is wrong because common.VERTICES has latitudes that are greater than 90 degrees.

Reply:
Through out other tests subscriptions are created in the same Area, that is the reason why I used 'any' instead of specific index, but what I can do is to make sure I clean up other test making sure there are zero subs until it gets to this one

Reply:
I its probably easier if I just picked a different location, so we don't have to worry about cleaning it up if we add more test in the future.

Reply:
That sounds good -- in general, we should make sure that tests don't interfere with each other.  It seems like the easiest way to accomplish that would be to have each test clean up after itself, but ensuring that every area-using test has its own area would work too.